backend/mutations.py

```diff
@@ -1,58 +1,78 @@
 import graphene
-
-from models import User, UserMetrics, Portfolio, Settings, Achievement, Group
+from bson import ObjectId
+from models import User, UserMetrics, Portfolio, Settings, Achievement, Group, Artwork, ArtworkMetrics
 from api_types import UserType, UserMetricsType, PortfolioType, SettingsType, AchievementType, GroupType
+from api_types import ArtworkType
 
 '''
-List of possible mutations:
-    create_user
-    update_user - need to add update metrics and settings
-    delete_user
+Need to research more into mutation arguments, but I think how I have them right now is how it should be
+List of possible mutations to make:
 
     create_group - todo
     update_group - todo
-    delete_group - todo, when should a group be deleted?
+    delete_group - todo
 
-    
+    create_achievement - todo
+    update_achievement - todo
+    remove_achievement - todo
+
+    incrementing metrics mutations (users and artworks) - todo    
 '''
 
+# NOTE: reference fields need ObjectId-typed input
+# NOTE: Collection methods - Remove deletes everything in the list, pop removes by index
+# Idea, might wanna store the graphql id in mongodb or vise versa somehow
+
+class UserSettingsInput(graphene.InputObjectType):
+    autoAddToGroupPortfolio = graphene.Boolean()
 
+class UserMetricsInput(graphene.InputObjectType):
+    works_visited = graphene.Int()
+    works_found = graphene.Int()
+    cities_visited = graphene.Int()
+    posts_written = graphene.Int()
 
 class UserInput(graphene.InputObjectType):
-    # id = graphene.ID()
-    name = graphene.String(required=True)
+    # since this class is generalized for all updating and creating, won't make any attributes required
+    id = graphene.String()
```

Comment:
This should be created on the backend. If we passed this from the frontend it would likely conflict with existing ids. Use the auto-generated ids that come with mongo engine

Reply:
I use it here as input when editing a current user in order to fetch that user from the database (So it would have to be a mongo object id). I can try to think of a different way though with graphQL's different IDs in mind or having it as a separate argument as to not confuse the two.

Reply:
Cool that makes sense, I think the type may be graphene.ID though